SpatialGDK/Source/SpatialGDK/Public/Interop/LoadBalancingWriter.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/ActorGroupMember.h"
+#include "Schema/ActorSetMember.h"
+#include "SpatialCommonTypes.h"
+#include "SpatialView/EntityComponentTypes.h"
+#include "UObject/WeakObjectPtr.h"
+
+#include "SpatialView/SubView.h"
+
+class USpatialNetDriver;
+
+namespace SpatialGDK
+{
+class FLoadBalancingWriterBase
+{
+public:
+	FLoadBalancingWriterBase(USpatialNetDriver* InNetDriver)
+		: NetDriver(InNetDriver)
+	{
+	}
+
+	const EntityViewElement& GetEntityView(Worker_EntityId EntityId) const;
+	void SendComponentUpdate(Worker_EntityId EntityId, FWorkerComponentUpdate& ComponentUpdate);
+
+protected:
+	TWeakObjectPtr<USpatialNetDriver> NetDriver;
```

Comment:
If these classes have the same lifecycle as the NetDriver, a Raw pointer should be enough.

Reply:
Isn't there an period of time where a UObject shouldn't be accessed (as in, it's effectively invalid) after it's PendingKill but before it's destroyed in a GC pass? It's rather defensive, I know.

Reply:
True, but we are dealing with the NetDriver and its components, not AActors in the wild trying to chat together. I would hope that before trying to destroy the net driver, things have settled a bit and nothing is trying to tick the net driver. And nothing outside the net driver is supposed to be using these classes.